cmsplugin_cascade/plugin_base.py

```diff
@@ -294,15 +294,24 @@ def save_model(self, request, new_obj, form, change):
                     new_obj.glossary[key] = old_obj.glossary[key]
         super(CascadePluginBase, self).save_model(request, new_obj, form, change)
 
-    def get_parent_instance(self, request=None):
+    def get_parent_instance(self, request=None, obj=None):
         """
         Get the parent model instance corresponding to this plugin. When adding a new plugin, the
         parent might not be available. Therefore as fallback, pass in the request object.
         """
-        try:
-            parent_id = self.parent.id
-        except AttributeError:
-            parent_id = request.GET.get('plugin_parent') if request else None
+        if obj is not None and hasattr(obj, 'parent_id'):
```

Comment:
Thanks for your advices, explanations and the link to EAFP.

So I have rewrite : 

        try:
            parent_id = obj.parent_id
        except AttributeError:
            try:
                parent_id = self.parent.id
            except AttributeError:
                if request:
                    parent_id = request.GET.get('plugin_parent', None)
                    if parent_id is None:
                        from cms.models import CMSPlugin
                        parent_id = CMSPlugin.objects.filter(id=request.resolver_match.args[0]
                                                             ).only("parent_id").order_by('?').first().parent_id
                else:
                    parent_id = None

But before to update the commits of the PR, I have tested the UI with django-cms 3.4.1 and I got no error in the code, but in the comportement of the UI when I make a copy-paste of a row containing a column, containing a text (ckeditor) containing a cms plugin  (either cascade plugin or filer image).

So I ran the tests you provided with cms cascade but with the following change in requirements : 
django-cms==3.4.1

And I got lots of errors :-(

So before to commit new changes, I will first check the source of the errors this week-end or next week.

Kr,

Patrick